bokeh/models/annotations/html/labels.py

```diff
@@ -0,0 +1,329 @@
+
+    '''
+
+    x = NumberSpec(default=field("x"), help="""
+    The x-coordinates to locate the text anchors.
+    """)
+
+    x_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the ``xs`` attribute. Interpreted as |data units| by
+    default.
+    """)
+
+    y = NumberSpec(default=field("y"), help="""
+    The y-coordinates to locate the text anchors.
+    """)
+
+    y_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the ``ys`` attribute. Interpreted as |data units| by
+    default.
+    """)
+
+    text = StringSpec(default=field("text"), help="""
+    The text values to render.
+    """)
+
+    angle = AngleSpec(default=0, help="""
+    The angles to rotate the text, as measured from the horizontal.
+    """)
+
+    x_offset = NumberSpec(default=0, help="""
+    Offset values to apply to the x-coordinates.
+
+    This is useful, for instance, if it is desired to "float" text a fixed
+    distance in |screen units| from a given data position.
+    """)
+
+    y_offset = NumberSpec(default=0, help="""
+    Offset values to apply to the y-coordinates.
+
+    This is useful, for instance, if it is desired to "float" text a fixed
+    distance in |screen units| from a given data position.
+    """)
+
+    text_props = Include(TextProps, help="""
+    The {prop} values for the text.
+    """)
+
+    background_props = Include(FillProps, prefix="background", help="""
+    The {prop} values for the text bounding box.
+    """)
+
+    background_fill_color = Override(default=None)
+
+    border_props = Include(LineProps, prefix="border", help="""
+    The {prop} values for the text bounding box.
+    """)
+
+    border_line_color = Override(default=None)
+
+class Title(HTMLAnnotation, Qualified):
+    ''' Render a single title box as an annotation.
+
+    See :ref:`userguide_annotations_titles` for information on plotting titles.
+
+    '''
+
+    text = String(default="", help="""
+    The text value to render.
+    """)
+
+    vertical_align = Enum(VerticalAlign, default='bottom', help="""
+    Alignment of the text in its enclosing space, *across* the direction of the text.
+    """)
+
+    align = Enum(TextAlign, default='left', help="""
+    Alignment of the text in its enclosing space, *along* the direction of the text.
+    """)
+
+    text_line_height = Float(default=1.0, help="""
+    How much additional space should be allocated for the title. The value is provided
+    as a number, but should be treated as a percentage of font size. The default is
+    100%, which means no additional space will be used.
+    """)
+
+    offset = Float(default=0, help="""
+    Offset the text by a number of pixels (can be positive or negative). Shifts the text in
+    different directions based on the location of the title:
+
+        * above: shifts title right
+        * right: shifts title down
+        * below: shifts title right
+        * left: shifts title up
```

Comment:
cc @tcmetzger for visibility, things like this will  need updating if/when we support RTL text. Maybe a more succinct way of expressing the affect of `offset` would be to say the offset shifts the beginning of the text in the direction of the text, and not that currently only LTR text is supported.